src/Dev/Tasks/MigrateFileTask.php

```diff
@@ -22,36 +23,127 @@ class MigrateFileTask extends BuildTask
-            DB::alteration_message("No file migration helper detected", "notice");
-            return;
+        $args = $request->getVars();
+        $this->validateArgs($args);
+
+        $subtasks = !empty($args['only']) ? explode(',', $args['only']) : $this->defaultSubtasks;
+
+        if (in_array('move-files', $subtasks)) {
+            if (!class_exists(FileMigrationHelper::class)) {
+                $this->logger->error("No file migration helper detected");
+                return;
+            }
+
+            $this->logger->info('### Migrating filesystem and database records (move-files)');
+
+            $this->logger->info('If the task fails or times out, run it again and it will start where it left off.');
+
+            $migrated = FileMigrationHelper::singleton()->run();
+            if ($migrated) {
+                $this->logger->info("{$migrated} File DataObjects upgraded");
+            } else {
+                $this->logger->info("No File DataObjects need upgrading");
+            }
         }
 
-        DB::alteration_message(
-            'If the task fails or times out, run it again and it will start where it left off.',
-            "info"
-        );
+        if (in_array('move-thumbnails', $subtasks)) {
+            if (!class_exists(LegacyThumbnailMigrationHelper::class)) {
+                $this->logger->error("LegacyThumbnailMigrationHelper not found");
+                return;
+            }
+
+            $this->logger->info('### Migrating existing thumbnails (move-thumbnails)');
 
-        $migrated = FileMigrationHelper::singleton()->run();
-        if ($migrated) {
-            DB::alteration_message("{$migrated} File DataObjects upgraded", "changed");
-        } else {
-            DB::alteration_message("No File DataObjects need upgrading", "notice");
+            $moved = LegacyThumbnailMigrationHelper::singleton()
+                ->setLogger($this->logger)
+                ->run($this->getStore());
+
+            if ($moved) {
+                $this->logger->info(sprintf("%d thumbnails moved", count($moved)));
+            } else {
+                $this->logger->info("No thumbnails moved");
+            }
         }
 
-        if (!class_exists(ImageThumbnailHelper::class)) {
-            DB::alteration_message("No image thumbnail helper detected", "notice");
-            return;
+        if (in_array('generate-cms-thumbnails', $subtasks)) {
+            if (!class_exists(ImageThumbnailHelper::class)) {
+                $this->logger->error("ImageThumbnailHelper not found");
+                return;
+            }
+
+            $this->logger->info('### Generating new CMS UI thumbnails (generate-cms-thumbnails)');
+
+            ImageThumbnailHelper::singleton()->run();
+        }
+    }
+
+    public function getDescription()
+    {
+        return <<<TXT
+Imports all files referenced by File dataobjects into the new Asset Persistence Layer introduced in 4.0.
```

Comment:
This only gets displayed in the web interface, you need HTML in there for it to display in a useful fashion.

Reply:
True that, I've shortened it a bit. Didn't add HTML to the link, but I think that's OK.